intelmq/bots/experts/recordedfuture_iprisk/expert.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+"""
+See README for database download.
+"""
+
+from intelmq.lib.bot import Bot
+import csv
+
+
+class RecordedFutureIPRiskExpertBot(Bot):
+
+    database = dict()
+
+    def init(self):
+        self.logger.info("Loading recorded future risk list.")
+
+        try:
+            with open(self.parameters.database) as fp:
+                rfreader = csv.DictReader(fp)
+                for row in rfreader:
+                    self.database[row['Name']] = row['Risk']
+
+        except IOError:
+            raise ValueError("Recorded future risklist not defined or failed on open.")
+
+        self.overwrite = getattr(self.parameters, 'overwrite', False)
+
+    def process(self):
+        event = self.receive_message()
+
+        for key in ["source", "destination"]:
+            if key + '.ip' in event:
+                if "extra.rf_iprisk." + key not in event:
+                    if event.get(key + '.ip') in self.database:
+                        event.add("extra.rf_iprisk." + key, int(self.database[event.get(key + '.ip')]))
```

Comment:
You can use `self.database.get(event.get(key + '.ip'), 0)` here instead of the four lines. Alternatively: [defaultdict](https://docs.python.org/3/library/collections.html?highlight=defaultdict#collections.defaultdict) :)